minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java

```diff
@@ -53,10 +55,26 @@ public void setMinikube(String minikube) {
   /**
    * Gets the minikube docker environment variables by running the command 'minikube docker-env
    * --shell=none'.
+   *
+   * @return A map of docker environment variables and their values
    */
   public Map<String, String> getDockerEnv() throws IOException, InterruptedException {
+    return getDockerEnv("");
+  }
+
+  /**
+   * Gets the minikube docker environment variables by running the command 'minikube docker-env
+   * --shell=none'.
+   *
+   * @param profile Target minikube profile
+   * @return A map of docker environment variables and their values
+   */
+  public Map<String, String> getDockerEnv(String profile) throws IOException, InterruptedException {
+    Preconditions.checkNotNull(profile, "Minikube profile must not be null");
+
     List<String> minikubeDockerEnvCommand =
-        Arrays.asList(minikube.get(), "docker-env", "--shell=none");
+        Arrays.asList(minikube.get(), "docker-env", "--shell=none", "--profile=".concat(profile));
```

Comment:
nit: I think just `"--profile=" + profile`gives us better readbility. Same for the other places.